src/android/FileTransfer.java

```diff
@@ -316,7 +317,23 @@ public void run() {
                     conn.setRequestProperty("Content-Type", "multipart/form-data; boundary=" + BOUNDARY);
 
                     // Set the cookies on the response
-                    String cookie = CookieManager.getInstance().getCookie(target);
+                    boolean gotCookie = false;
+                    String cookie = null;
+                    Class webViewClass = webView.getClass();
+                    try {
+                        Method gcm = webViewClass.getMethod("getCookieManager");
+                        ICordovaCookieManager cm = (ICordovaCookieManager)gcm.invoke(webView);
```

Comment:
We'll need to make the code just a little be more complicated and hard to read still, since pre-4.0.0 cordova-android users will get a compile error looking for the ICordovaCookieManager symbol :(